chain/chain/src/runtime/mod.rs

```diff
@@ -849,18 +850,31 @@ impl RuntimeAdapter for NightshadeRuntime {
                 }
 
                 let signer_id = validated_tx.signer_id();
-                let (signer, access_key) = if let Some((id, signer, key)) = &mut signer_access_key {
-                    debug_assert_eq!(signer_id, id);
-                    (signer, key)
+                let payer_id = match validated_tx.key() {
```

Comment:
have you considered having 2 separate functions for account and access keys here? there are seems to be very little overlap and it looks like at every step we have different code path